.github/workflows/test_deployment.yml

```diff
@@ -1,13 +1,21 @@
 name: Deployment Test
-on: workflow_dispatch
+on:
+    push:
+      branches: [ '*' ]
 jobs:
     main:
-        name: Build package with Poetry and run example Constrin script
-        runs-on: ['ubuntu-latest']
+        name: Build package with Poetry and run example Constrain script
+        runs-on: ${{ matrix.os }}
+        strategy:
+            matrix:
+                os: [ubuntu-latest] #, windows-latest, macos-latest]
+                python-version: ["3.10"]
```

Comment:
Only running on `ubuntu` for now but the workflow was also successful on `windows` and `macos`.